Cabal/Distribution/Simple/SrcDist.hs

```diff
@@ -216,7 +216,7 @@ listPackageSourcesOrdinary verbosity pkg_descr pps =
   , fmap concat
     . withAllLib $ \ l -> do
        let lbi = libBuildInfo l
-           relincdirs = "." : filter isRelative (includeDirs lbi)
+           relincdirs = "." : filter isRelativeOnAnyPlatform (includeDirs lbi)
```

Comment:
Probably no benefit. At least no scenario which is easy to come up with and seems very likely.